umake/ui/cli/__init__.py

```diff
@@ -157,6 +158,65 @@ def mangle_args_for_default_framework(args):
     return result_args
 
 
+def get_frameworks_list(args):
+    """
+    Get a frameworks list based on the arguments.
+    Multiple forms of the frameworks list can ge given:
+        - List with all frameworks
+        - List with just only installed frameworks
+        - List with just installable frameworks
+
+    Args:
+        args: It's the return of parse_args function. Just args.list, args.list_available and args.list_installed
+        are need to this function.
+
+    Return:
+        Give the frameworks list as a string
+    """
+    categories = list_frameworks()
+    print_result = str()
+
+    if args.list or args.list_available:
+        # Sort the categories to prevent a random list at each new program execution
+        for category in sorted(categories, key=lambda cat: cat["category_name"]):
+            if category["is_installed"] == 0:
+                cat_is_installed = _("not installed")
+            elif category["is_installed"] == 1:
+                cat_is_installed = _("partially installed")
+            else:
+                cat_is_installed = _("fully installed")
+
+            print_result += "{}: {} ({})\n".format(category["category_name"],
+                                                   category["category_description"],
+                                                   cat_is_installed)
+
+            # Sort the frameworks to prevent a random list at each new program execution
+            for framework in sorted(category["frameworks"], key=lambda fram: fram["framework_name"]):
+                if args.list_available:
+                    if not framework["is_installable"]:
+                        continue
+
+                print_result += "\t{}: {}".format(framework["framework_name"], framework["framework_description"])
+
+                if not framework["is_installable"]:
+                    print_result = _("{} ({})".format(print_result, _("not installable on this machine")))
```

Comment:
actually, this could be:
`print_result = _("{} (not installable on this machine)".format(print_result))`

and same for below, making sense? (sorry, was tired yesterday when giving this example ;))